src/components/DetailView/DetailView.svelte

```diff
@@ -59,11 +59,40 @@
       }).then((rows) => addMissing(rows, sensor))
     : [];
 
+  // The currently selected detailViewTimeSpan, initially defaults to the smallMultipleTimeSpan.
+  // Modified by onDateRangeChange, and fetched via (non-reactive) getDateRange,
+  // to avoid updating the chart immediately, so it will be used as the initial value the
+  // next time createSpec is called.  Reloading the component will reset to the default.
+  $: detailViewTimeSpan = detailViewTimeSpan || $smallMultipleTimeSpan;
+
+  function getDateRange() {
+    if (detailViewTimeSpan.length === 0) {
+      detailViewTimeSpan = $smallMultipleTimeSpan;
+    }
+    return detailViewTimeSpan;
+  }
+
+  function onDateRangeChange(event) {
+    // All signal events are sent to all on:signal handlers (??), so check if this is the right handler.
+    if (event.detail.name !== 'dateRange') {
+      return;
+    }
+    const dr = event.detail.value && event.detail.value.date_value;
+    if (dr && Array.isArray(dr) && dr.length > 0) {
+      if (
+        (!Number.isNaN(dr[0]) && dr[0] !== detailViewTimeSpan[0].getTime()) ||
+        (!Number.isNaN(dr[1]) && dr[1] !== detailViewTimeSpan[1].getTime())
+      ) {
```

Comment:
Maybe we shouldn't bother checking whether the value has changed, since it won't be used immediately anyway.

But the check that both values are not NaN is important anyway, since NaNs will be sent when there is no data for the sensor.